drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -5044,9 +5044,9 @@ Vector<uint8_t> RenderingDeviceVulkan::shader_compile_binary_from_spirv(const Ve
 				ERR_FAIL_COND_V_MSG(result != SPV_REFLECT_RESULT_SUCCESS, Vector<uint8_t>(),
 						"Reflection of SPIR-V shader stage '" + String(shader_stage_names[p_spirv[i].shader_stage]) + "' failed enumerating output variables.");
 
-				if (ov_count) {
+				if (ov_count > 0) {
 					Vector<SpvReflectInterfaceVariable *> output_vars;
-					output_vars.resize(ov_count);
+					output_vars.resize_zeroed(ov_count);
```

Comment:
This seems fishy to me. ``spvReflectEnumerateOutputVariables`` enumerates over ``ov_count`` and fills ``output_vars`` _and_ even if it didn't totally cover ``output_vars`` there is the nullptr check before accessing the contents of refvar, so it should be safe. 

Am I missing something?

```
if (IsNotNull(pp_variables)) {
    if (*p_count != p_module->output_variable_count) {
      return SPV_REFLECT_RESULT_ERROR_COUNT_MISMATCH;
    }

    for (uint32_t index = 0; index < *p_count; ++index) {
      SpvReflectInterfaceVariable* p_var = p_module->output_variables[index];
      pp_variables[index] = p_var;
    }
  }
  else {
    *p_count = p_module->output_variable_count;
  }
```